action.yml

```diff
@@ -138,6 +138,18 @@ runs:
         export GITHUB_ACTION_PATH=${{ github.action_path }}
         export ACTION_REPOSITORY="WIPACrepo/wipac-dev-py-setup-action"
 
+        # get python3 version (max)
+        export SEMVAR_RANGE=$(grep "python_requires =" -h setup.cfg | awk -F 'python_requires =' '{print $2}')
+        export LATEST_PYTHON3=$(python3 ${{ github.action_path }}/latest_python3.py)
+        pip3 install semantic-version
+        export PACKAGE_MAX_PYTHON_VERSION=$(python -c '
+        import semantic_version, os
+        spec = semantic_version.SimpleSpec(os.getenv("SEMVAR_RANGE").replace(" ",""))
+        latest_py3_minor = int(os.getenv("LATEST_PYTHON3").split(".")[1])
+        maxo = max(spec.filter(semantic_version.Version(f"3.{i}.0") for i in range(latest_py3_minor + 1)))
+        print(f"{maxo.major}.{maxo.minor}")
+        ')
```

Comment:
That's a lot of python to shove into a string.  I'd almost prefer it to be a helper script like above.  Or just integrate everything into one script, if you don't need `LATEST_PYTHON3` separate.

Reply:
Good point, I'll do that